### ISSUE TITLE
[SUGGESTION]: Setting to remove "Leave Dungeon" button

### ISSUE THREAD

**Issue by M-Cool** [pokeclicker/pokeclicker#5333]
### Current Behavior

If you return to the starting position of a dungeon (Blue square) after moving out of it, or a Dungeon Guide auto-restarts a dungeon (v0.10.20), the main module will have a button with the text "Leave Dungeon". 

If you click any pixel of the main module or use the "Interact" hotkey, when that button is on-screen, it will create a pop-up window to confirm that you want to leave that doesn't allow interacting with anything until you click accept (leave the dungeon) or cancel.

The only alternative to this behavior is a setting that auto-accept the pop-up, exiting the dungeon immediately. No option to disable this mechanic is given to the player.

### Improved Behavior

Modify the 4th Setting of "Settings > Gameplay": "Confirm before leaving dungeons: Yes // No checkbox" to be a 3 option list, similarly to the 1st setting of that page "Shop amount buttons: +10,+100 // +100,+1000 // x10, ·/·10).

![imagen](https://github.com/pokeclicker/pokeclicker/assets/143295031/c74fde43-1361-4f96-92c7-71ab7c2b998b)

The first 2 options would be the same that the current ON / OFF checkmark allows choosing, while the 3rd option (the only new mechanic) will be disabling the current behavior entirely (make the button never appear, until the setting is changed again). 

### Reason for change

Prior to v0.10.20, the only way that this could affect players is if they were constantly clicking while moving, either to open chests as fast as possible or because they were speedrunning low-difficulty dungeons where click attack could help to move faster, and passed through the blue tile again. A good example of this phenomena being Orre1 where due to low-hp trainers clicking when possible is greatly incentivized. 

While that little annoyance would be more than enough to consider giving the option to ignore this mechanic (leaving the dungeon is usually more trouble than it is worth), the main problem that should be addressed is that after the v0.10.20, due to a small bug related to the Dungeon Guide, if a guide auto-resets the dungeon, the "Leave the Dungeon" button will already be there even without moving from the starting position.
![Leave_Dungeon](https://github.com/pokeclicker/pokeclicker/assets/143295031/cf87e83f-f96b-489f-94c3-e5db9c203fe3)

This is quite a bit more problematic since it means that if you are using click attacks during the boss fight (because it's a trainer, or you need the DPS because the guide took a bit too long to reach it) there is a big chance that you will instead click the button (that you couldn't even see).  Not only this is annoying (you will need to take a bit of time to cancel every time it happens) but in the worst case scenario you could accidentally click the accept button (do an extra click without noticing it), not only leaving the dungeon but also losing ALL the money that you pay upfront to the guide (up to Billions of P$ and DT if you pay to try 500 attempts of certain dungeons).

**TLDR: The "leave dungeon" button is usually more trouble (it's a constant danger that you need to actively keep in mind if you click the main module often) than it is worth (its only use is farming items, in dungeons with 500 clears, slightly faster). Giving options to players it's usually a good practice since it allows everyone to play like they want (instantly leaving to farm items faster or not having to worry about accidentally leaving when farming the 500 clears). Even if this suggestion is partially rejected, something must be done to fix the instant pop-up when using the dungeon guides since it's not only really common but also is when accidentally leaving is more costly (you lose all your upfront payment to the guide)**

### Detailed Description

The name of this new setting would be something like "**Allow leaving dungeons by returning to the start**" and the 3 options would be: 
**1-Leave immediately**: Equivalent to what happens currently if you do NOT activate the checkbox (clicking when the prompt is on the screen allows leaving without needing to confirm it).

**2-Leave after a warning**: Equivalent to what happens currently if you activate the checkbox (clicking creates the confirmation pop-up before actually leaving). If a 3 choice setting it's not possible this "Leave after warning" should probably be eliminated in order to at least keep option 1 (leaving as fast as possible) and option 3 (not having to worry about leaving)

**3-Require to timeout**: Only new mechanic where the blue tile (spawning position) is just a regular white tile with no button that allow you to exit from the dungeon, meaning that the only way to exit is by waiting around 60 seconds (losing via a timeout).

**Comment by CypherX** [comment]
My recommendation would be to just improve the logic for when the Leave Dungeon button appears so a guide auto-restarting the dungeon doesn't trigger it to display immediately
